File: BooleanMatrixJava/src/booleanmatrixjava/Matrix.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package booleanmatrixjava;

/**
 *
 * @author Johann
 */
public class Matrix 
{
    public boolean[][] matrix;
    public int rowCount;
    public int columnCount;
    
    Matrix(int rows, int cols)
    {
        rowCount = rows;
        columnCount = cols;
        matrix = new boolean[rows][cols];
    }
    
    Matrix(boolean[][] data)
    {
        this.matrix = data; 
        this.rowCount = data.length;
        this.columnCount = data[0].length;
    }
    
    private boolean validOperation(Matrix m)
    {
        return m.rowCount == this.rowCount && m.columnCount == this.columnCount;
    }
    
    public Matrix getUnion(Matrix m1, Matrix m2)
    {
        if(m1.validOperation(m2))
        {
            Matrix result = new Matrix(m1.rowCount, m1.columnCount);
            for (int i = 0; i < m1.rowCount; i++) 
            {
                for (int j = 0; j < m1.columnCount; j++) 
                {
                    result.matrix[i][j] = m1.matrix[i][j] || m2.matrix[i][j];
                }
            }
            return result;
        }
        else
        {
            return null;
        }
    }
    
    public Matrix getConjunction(Matrix m1, Matrix m2)
    {
        if(m1.validOperation(m2))
        {
            Matrix result = new Matrix(m1.rowCount, m1.columnCount);
            for (int i = 0; i < m1.rowCount; i++) 
            {
                for (int j = 0; j < m1.columnCount; j++) 
                {
                    result.matrix[i][j] = m1.matrix[i][j] && m2.matrix[i][j];
                }
            }
            return result;
        }
        else
        {
            return null;
        }
    }
    
    public Matrix getInverse(Matrix m)
    {
        Matrix result = new Matrix(m.rowCount,m.columnCount);
        for (int i = 0; i < m.rowCount; i++) 
        {
            for (int j = 0; j < m.columnCount; j++) 
            {
                result.matrix[i][j] = !m.matrix[i][j];
            }
        }
        return result;
    }
    
    public Matrix getComplement(Matrix m)
    {
        Matrix result = new Matrix(m.rowCount,m.columnCount);
        for (int i = 0; i < m.rowCount; i++) 
        {
            for (int j = 0; j < m.columnCount; j++) 
            {
                result.matrix[i][j] = !m.matrix[i][j];
            }
        }
        return result;
    }
    
    public Matrix getComposition(Matrix m1, Matrix m2)
    {
        if(m1.columnCount == m2.rowCount)
        {
            Matrix result = new Matrix(m1.rowCount, m2.columnCount);
            for (int i = 0; i < m1.rowCount; i++) 
            {
                for (int j = 0; j < m2.columnCount; j++) 
                {
                    boolean eq = false;
                    for (int k = 0; k < m1.columnCount; k++) 
                    {
                        eq = eq || (m1.matrix[i][k] && m2.matrix[k][j]);
                    }
                    result.matrix[i][j] = eq;
                }
                
            }
            return result;
        }
        else
        {
            return null;
        }
    }
    
    public void print()
    {
        for(int i=0; i<this.rowCount;i++)
        {
            System.out.print("| ");
            for(int j=0; j<this.columnCount;j++)
            {
                String data = this.matrix[i][j] ? "1" : "0";
                System.out.print(data+" ");
            }
            System.out.println("|");
        }
    }
}

File: BooleanMatrixCalculator/src/Matrix.cpp
#include "Matrix.h"
#include <iostream>

using namespace std;

Matrix::Matrix(int c, int r)
{
    this->rows = r;
    this->cols = c;
    system("cls");
    cout << "Ingrese 1 o 0 segun corresponda en la posicion\n";
    for(int i=0; i<rows;i++)
    {
        for(int j=0;j<rows;j++)
        {
            int data;
            do
            {
                cout << "Ingrese el valor en la fila" << i << ", columna "<<j<<": ";
                cin >> data;
                if(data!=1 && data!=0)
                {
                    cout << "Solo ingrese 1 o 0\n";
                }
            }while(data!= 1 && data!= 0);
            matrix[i][j] = (bool)data;
        }
    }
}

void Matrix::print()
{
    for(int i=0; i<rows;i++)
    {
        cout << "| ";
        for(int j=0; j<cols;j++)
        {
            cout << matrix[i][j]<< " ";
        }
        cout << "|" << endl;
    }
}

bool Matrix::isValid(Matrix m1)
{
    return m1.cols == this->cols && m1.rows == this->rows;
}

void Matrix::getUnion(Matrix &m1, Matrix &m2)
{
    if(m1.isValid(m2))
    {
        this->rows = m1.cols;
        this->cols = m1.rows;
        for(int i=0; i<this->rows;i++)
        {
            for(int j=0; j<this->cols;j++)
            {
                this->matrix[i][j] = (bool) m1.matrix[i][j] || m2.matrix[i][j];
            }
        }
    }
}

void Matrix::getComposition(Matrix &m1, Matrix &m2)
{
    if(m1.cols == m2.rows)
    {
        this->rows = m1.rows;
        this->cols = m2.cols;
        for(int i=0; i<this->rows; i++)
        {
            for(int j=0;j<this->cols;j++)
            {
                bool eq = false;
                for(int k=0; k<m1.cols;k++)
                {
                    eq = eq || (m1.matrix[i][k] && m2.matrix[k][j]);
                }
                this->matrix[i][j] = eq;
            }
        }
    }
}

void Matrix::getComplement(Matrix &m)
{
    this->rows = m.rows;
    this->cols = m.cols;
    for(int i=0;i<rows;i++)
    {
        for(int j=0;j<cols;j++)
        {
            this->matrix[i][j] = !m.matrix[i][j];
        }
    }
}

void Matrix:: getConjunction(Matrix &m1, Matrix &m2)
{
    if(m1.isValid(m2))
    {
        this->rows = m1.cols;
        this->cols = m1.rows;
        for(int i=0; i<this->rows;i++)
        {
            for(int j=0; j<this->cols;j++)
            {
                this->matrix[i][j] = (bool) m1.matrix[i][j] && m2.matrix[i][j];
            }
        }
    }
}

Matrix::~Matrix()
{
}

Matrix::Matrix()
{
    this->rows = 0;
    this->cols = 0;
}
